pkg/controller/elasticsearch/client/model.go

```diff
@@ -104,97 +103,35 @@ type ClusterStateNode struct {
 	} `json:"attributes"`
 }
 
-// Shards contains the shards in the Elasticsearch routing table
-// mapped to their shard number.
-type Shards struct {
-	Shards map[string][]Shard `json:"shards"`
-}
+// Shards contains the shards in the Elasticsearch cluster
+type Shards []Shard
 
-// ClusterState partially models Elasticsearch cluster state.
-type ClusterState struct {
-	ClusterName  string                      `json:"cluster_name"`
-	ClusterUUID  string                      `json:"cluster_uuid"`
-	Version      int                         `json:"version"`
-	MasterNode   string                      `json:"master_node"`
-	Nodes        map[string]ClusterStateNode `json:"nodes"`
-	RoutingTable RoutingTable                `json:"routing_table"`
+// Shard partially models Elasticsearch cluster shard.
+type Shard struct {
+	Index  string `json:"index"`
+	Shard  string `json:"shard"`
+	Prirep string `json:"prirep"`
+	State  string `json:"state"`
+	// Node is the node name not the Node id
+	Node string `json:"node"`
 }
 
 type RoutingTable struct {
 	Indices map[string]Shards `json:"indices"`
 }
 
-// IsEmpty returns true if this is an empty struct without data.
-func (cs ClusterState) IsEmpty() bool {
-	return cs.ClusterName == "" &&
-		cs.ClusterUUID == "" &&
-		cs.Version == 0 &&
-		cs.MasterNode == "" &&
-		len(cs.Nodes) == 0 &&
-		len(cs.RoutingTable.Indices) == 0
-}
-
-// GetShards reads all shards from cluster state,
-// similar to what _cat/shards does but it is consistent in
-// its output.
-func (cs ClusterState) GetShards() []Shard {
-	var result []Shard
-	for _, index := range cs.RoutingTable.Indices {
-		for _, shards := range index.Shards {
-			for _, shard := range shards {
-				shard.Node = cs.Nodes[shard.Node].Name
-				result = append(result, shard)
-			}
-		}
-	}
-	return result
-}
-
 // GetShardsByNode returns shards by node.
 // The result is a map with the name of the nodes as keys and the list of shards on the nodes as values.
-func (cs ClusterState) GetShardsByNode() map[string][]Shard {
+func (s Shards) GetShardsByNode() map[string][]Shard {
 	result := make(map[string][]Shard)
-	for _, index := range cs.RoutingTable.Indices {
-		for _, shards := range index.Shards {
-			// for each shard, check if it assigned to a node
-			for _, shard := range shards {
-				if len(shard.Node) == 0 {
-					continue
-				}
-				// shard.Node is the id of the node, get the corresponding node name
-				nodeName := cs.Nodes[shard.Node].Name
-				result[nodeName] = append(result[nodeName], shard)
-			}
+	for _, shard := range s {
+		if len(shard.Node) > 0 {
```

Comment:
Does this happen when shards are unassigned? I wonder if dropping them is the right decision, rather adding them to the `""` key's values.

Reply:
They would not be used, not sure if it's worth it to change this in this PR.
But your point is interesting: should we take care of `UNASSIGNED` shards ?

Reply:
It could be good to add a comment to not forget that `UNASSIGNED` shards are ignored because they have no node?